frontend/src/components/viewers/Tensorboard.tsx

```diff
@@ -70,51 +89,167 @@ class TensorboardViewer extends Viewer<TensorboardViewerProps, TensorboardViewer
+                  <MenuItem value={'1.8.0'}>TensorFlow 1.8.0</MenuItem>
+                  <MenuItem value={'1.9.0'}>TensorFlow 1.9.0</MenuItem>
+                  <MenuItem value={'1.10.0'}>TensorFlow 1.10.0</MenuItem>
+                  <MenuItem value={'1.11.0'}>TensorFlow 1.11.0</MenuItem>
+                  <MenuItem value={'1.12.0'}>TensorFlow 1.12.0</MenuItem>
+                  <MenuItem value={'1.13.2'}>TensorFlow 1.13.2</MenuItem>
+                  <MenuItem value={'1.14.0'}>TensorFlow 1.14.0</MenuItem>
+                  <MenuItem value={'1.15.0'}>TensorFlow 1.15.0</MenuItem>
+                  <ListSubheader>TensorFlow 2.x</ListSubheader>
+                  <MenuItem value={'2.0.0'}>TensorFlow 2.0.0</MenuItem>
+                </Select>
+              </FormControl>
+            </div>
+            <div>
+              <BusyButton
+                className={commonCss.buttonAction}
+                disabled={!this.state.tensorflowVersion}
+                onClick={this._startTensorboard.bind(this)}
+                busy={this.state.busy}
+                title={`Start ${this.props.configs.length > 1 ? 'Combined ' : ''}Tensorboard`}
+              />
+            </div>
+          </div>
         )}
       </div>
     );
   }
 
+  private _handleDeleteOpen = () => {
+    this.setState({ deleteDialogOpen: true });
+  };
+
+  private _handleDeleteClose = () => {
+    this.setState({ deleteDialogOpen: false });
+  };
+
   private _buildUrl(): string {
     const urls = this.props.configs.map(c => c.url).sort();
     return urls.length === 1 ? urls[0] : urls.map((c, i) => `Series${i + 1}:` + c).join(',');
   }
 
   private async _checkTensorboardApp(): Promise<void> {
     this.setState({ busy: true }, async () => {
-      const podAddress = await Apis.getTensorboardApp(this._buildUrl());
-      this.setState({ busy: false, podAddress });
+      const tensorboardInstance = await Apis.getTensorboardApp(this._buildUrl());
+      const tbaddress = tensorboardInstance.podAddress;
+      const tfversion = tensorboardInstance.tfVersion;
+      this.setState({ busy: false, podAddress: tbaddress, tensorflowVersion: tfversion });
     });
   }
 
   private async _startTensorboard(): Promise<void> {
     this.setState({ busy: true }, async () => {
-      await Apis.startTensorboardApp(encodeURIComponent(this._buildUrl()));
+      await Apis.startTensorboardApp(
+        encodeURIComponent(this._buildUrl()),
+        encodeURIComponent(this.state.tensorflowVersion),
+      );
       this.setState({ busy: false }, () => {
         this._checkTensorboardApp();
       });
     });
   }
+
+  private async _deleteTensorboard(): Promise<void> {
+    // delete the already opened Tensorboard, clear the podAddress recorded in frontend,
+    // and return to the select & start tensorboard page
+    this.setState({ busy: true }, async () => {
+      await Apis.deleteTensorboardApp(
+        encodeURIComponent(this._buildUrl()),
```

Comment:
Apis.deleteTensorboardApp is already calling `encodeURIComponent`, this is a mistake?

Reply:
I wrote this way because startTensorboard also call encodeURIComponent twice. Will remove both if proven not necessary.